### PR TITLE
Issue/#40 define schema for db with validation (new branch)

### LINKED ISSUE #40: Define schema for Task with validation and tests
### User Story: 

As a software developer, I want a schema-based solution for a task so that the application data can be modeled according to the specs.

### Acceptance Criteria:

1) solution fits the needs of the front-end requirements and specs
2) solution contains validations and restrictions where required
3) unit tests

task should have: (compulsory) task-name, user, (optional) location, priority, reminder date, actual date, description,

### PR DESCRIPTION
**Pull request by emzoo16**
Extending on #64 
Resolves #40

- [ x] The pull request is complete according to the following criteria:
  - [ x] Acceptance criteria have been met
  - [ x] The documentation is kept up-to-date
  - [ x] Comprehensive tests (if applicable) have been generated and all pass.
- [ x] The pull request describes the changes that have been made, and enough information is present in the description for any developer to understand what has changed
- [ ] Commmits have been squashed
- [x ] The pull request title is of the following format : {issue/bug...}/(Issue Number) - Name of issue. E.g bug/30-Fix-Project
- [x ] The description uses [github syntax](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) to link to the issue. E,g Resolves se701g2/Doto#{Number}
- [x ] At least two reviewers assigned. One of which must be the assigner of the issue.
- [ ] If there are merge conflicts, run git rebase as opposed to git merge with master.

### CODE REVIEW

File: doto-backend/test/task.test.js [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    picture: 'profile.png',
+    themePreference: 'dark'
+})
+
+process.env.TEST_SUITE = 'task-test';
+
+describe('Task Model Tests', () => {
+
+    before(async () => {
+        await mongoose.connect(`mongodb://localhost:27017/${process.env.TEST_SUITE}`, { useNewUrlParser: true}, (err) => {
```
Comment (lucycy98) [inline doto-backend/test/task.test.js]: hey could you please change the url to this
`mongodb://127.0.0.1:27017/${process.env.TEST_SUITE}`

File: doto-backend/test/user.test.js [hunk 2]
```diff
@@ -8,31 +8,60 @@ process.env.TEST_SUITE = 'user-test';
 
 describe('User Model Test', () => {
 
-    beforeEach(async () => {
-        await mongoose.connect(`mongodb://127.0.0.1:27017/${process.env.TEST_SUITE}`, { useNewUrlParser: true}, (err) => {
+    before(async () => {
+        await mongoose.connect(`mongodb://localhost:27017/${process.env.TEST_SUITE}`, { useNewUrlParser: true}, (err) => {
```
Comment (lucycy98) [inline doto-backend/test/user.test.js]: and here too 
`mongodb://127.0.0.1:27017/${process.env.TEST_SUITE}`

**Comment by emzoo16** [comment]
Made the changes :)

**Comment by lucycy98** [comment]
Omg sorry one more thing, could you squash your commits and just have one commit message following the correct format?

**Pull request merged**